client/scala/src/main/scala/ai/verta/blobs/dataset/PathBlob.scala

```diff
@@ -0,0 +1,124 @@
+package ai.verta.blobs.dataset
+
+import ai.verta.swagger._public.modeldb.versioning.model._
+
+import java.security.{MessageDigest, DigestInputStream}
+import java.io.{File, FileInputStream}
+
+import scala.collection.mutable.HashMap
+import scala.util.{Failure, Success, Try}
+
+/** Captures metadata about files
+ *  @param paths list of filepaths or directory paths
+ */
+ case class PathBlob(private val paths: List[String]) extends Dataset {
+  private val BufferSize = 8192
+
+  private val metadataList = paths.map(expanduser)
+  .flatMap((path: String) => processPath(new File(path)))
+  .map(metadata => metadata.path -> metadata)
+
+  protected var contents = HashMap(metadataList: _*)
+
+  override def equals(other: Any) = other match {
+    case other: PathBlob => contents.equals(other.contents)
+    case _ => false
+  }
+
+
+  /** Hash the file's content
+   *  From https://stackoverflow.com/questions/41642595/scala-file-hashing
+   *  @param path filepath
+   */
+  private def hash(file: File) = Try {
+     val buffer = new Array[Byte](BufferSize)
+     val messageDigest = MessageDigest.getInstance("MD5")
+
+     val dis = new DigestInputStream(
+       new FileInputStream(file),
+       messageDigest
+     )
+
+     try {
+       while (dis.read(buffer) != -1) {}
+     } finally {
+       dis.close()
+     }
+
+     // Convert to hexadecimal
+     messageDigest.digest.map("%02x".format(_)).mkString
+   }
+
+   /** Get the metadata of path.
+    *  @param file a file object, representing the path
+    *  @return a list of components of file under the path
+    */
+   private def processPath(file: File): List[FileMetadata] = {
```

Comment:
This can fail, so it should return a try. You implicitly can throw things, which is not documented in the function signature. A try will enforce this knowledge at compile time.

Reply:
Resolved.